themes/bootstrap3/js/check_item_statuses.js

```diff
@@ -61,27 +61,17 @@ VuFind.register('itemStatuses', function ItemStatuses() {
       el.querySelectorAll('.callnumber,.hideIfDetailed,.location').forEach((e) => e.classList.add('hidden'));
       var locationListHTML = "";
       for (var x = 0; x < result.locationList.length; x++) {
+        var availability = result.locationList[x].availability;
+        var statusIcon =
+          availability === 'true' ? 'available'
+            : availability === 'false' ? 'unavailable'
+              : availability;
+
         locationListHTML += '<div class="groupLocation">';
-        if (result.locationList[x].availability) {
-          locationListHTML += '<span class="text-success">'
-            + VuFind.icon("status-available")
-            + result.locationList[x].location
-            + '</span> ';
-        } else if (typeof(result.locationList[x].status_unknown) !== 'undefined'
-          && result.locationList[x].status_unknown
-        ) {
-          if (result.locationList[x].location) {
-            locationListHTML += '<span class="text-warning">'
-              + VuFind.icon("status-unknown")
-              + result.locationList[x].location
-              + '</span> ';
-          }
-        } else {
-          locationListHTML += '<span class="text-danger">'
-            + VuFind.icon("status-unavailable")
-            + result.locationList[x].location
-            + '</span> ';
-        }
+        locationListHTML += '<span class="' + availabilityClasses[availability] + '">'
```

Comment:
Should we have fallback behavior if `availabilityClasses[availability]` is undefined for some reason (either the AJAX handler returns an unexpected value, or the scripts setting the underlying data get broken)? It might be helpful to log something to the console, or at least avoid a Javascript error caused by an illegal array access.

Reply:
Or alternatively, would it make sense to just pass the class as part of the AJAX response, and move some of the logic currently added to results-scripts.phtml into GetItemStatuses.php instead?

Reply:
I think returning the class and icon in GetItemStatuses.php would be cleaner and is easier to customize. Also the whole availabilityClasses would be unnecessary (I wasn't really happy with how I added it).
We could also directly return the rendered HTML. We have customized the HTML a bit, so returning rendered HTML would be appreciated. :-)
@ThoWagen what do you think?

Regarding the priority: For now I would leave it as is. I think this is a somewhat individual desicion for the libraries, so maybe a little survey would be good to determine what should be used as default.

Also, if we move this to 11.0, I would change the true and false in Helper\Root\AvailabilityStatus:availabilityAsString() to more descriptive terms. Previously I reverted that change in case you wanted to include this PR in 10.1

Reply:
@GrothTobias I just remembered that we had this discussion already here https://github.com/vufind-org/vufind/pull/3585#pullrequestreview-2001180237 . So yes, I think moving the logic from JS to GetItemStatuses.php would make sense.

Reply:
I wasn't clear if server-side rendering of the location list is wanted, so the classes and icons for each location are now returned in `GetItemStatuses.php`. I added the icon names (not sure what to call it) to `Helper\Root\AvailabilityStatus.php`, similar to the classes. 
With this the field `availability` is not needed for locations but I'm unsure if I should remove it - it could be useful if someone wants to customize this in the JS.

Reply:
I think at this point I've gone ahead and moved this to the 11.0 milestone, so you can feel free to make larger changes if you like. I don't mind doing more server-side rendering in the AJAX handler as long as it properly accounts for the theme system (e.g. use a template to render where appropriate). Less Javascript code constructing HTML is probably a good thing, for readability and customization. :-)

I'm going to hold off on further review/testing until you're done making the changes you wish to make, so please let me know when you are ready for me to take a closer look again!

Reply:
The `GetItemStatuses` handler now returns rendered HTML. As mentioned in the `@TODO` there is still something to do but I don't have time for it next week.

Reply:
Thanks for the progress; since this is now scheduled for release 11, there's no huge rush to finish -- just please let me know when you're ready for another review.